src/domain/entities/schemas/page.schema.factory.ts

```diff
@@ -0,0 +1,17 @@
+import { Schema, SchemaObject } from 'ajv';
+
+export function buildPageSchema(
+  keyRef: string,
+  itemSchema: SchemaObject,
+): Schema {
+  return <SchemaObject>{
+    $id: keyRef,
+    type: 'object',
+    properties: {
+      count: { type: 'number' },
+      next: { type: 'string', nullable: true },
+      previous: { type: 'string', nullable: true },
+      results: { type: 'array', items: { $ref: itemSchema.$id } },
+    },
+  };
```

Comment:
This function is meant to automatize the creation of `Page<T>` schemas.

I changed the approach while developing the feature: at first, I tried to automatize the `$id` field, by just adding a suffix to the `$id` of the input `itemSchema`, something like:
```
return <SchemaObject>{
  $id: itemSchema.replace('.json', '-page.json'),
  ...
```
Using this approach the function would take just one parameter, because `keyRef` would not be necessary anymore. But later I took the opposite approach and added `keyRef` parameter to be explicitly set by the caller. Since this `$id` will be referenced in several places (by validators when registering the schema) I didn't want to rely on inner logic. A third option consists of exporting a function that calculates the `$id` but it seemed a bit overkill to me.

I'm good with the current approach but also open to feedback of course.